scipy/sparse/csgraph/_flow.pyx

```diff
@@ -0,0 +1,395 @@
+import numpy as np
+
+from scipy.sparse import csr_matrix, isspmatrix_csr
+
+cimport cython
+cimport numpy as np
+
+
+class MaximumFlowResult:
+    """Represents the result of a maximum flow calculation.
+
+    Attributes
+    ----------
+    flow_value : int
+        The value of the maximum flow.
+    residual : csr_matrix
+        The residual graph with respect to the maximum flow.
+    """
+
+    def __init__(self, flow_value, residual):
+        self.flow_value = flow_value
+        self.residual = residual
```

Comment:
If we are going to be returning a class instance, then adding a `__repr__` method would be good. This isn't too informative at the moment:
```
In [3]:  maximum_flow(graph, 0, 1)                                                             
Out[3]: <scipy.sparse.csgraph._flow.MaximumFlowResult at 0x11028ff28>
```

Reply:
Since `MaximumFlowResult.residual` is typically going to be large, that one won't fit well into a `__repr__`, but I included one using `flow_value`:

```python
>>> maximum_flow(graph, 0, 1)
MaximumFlowResult with value of 5
```

Let me know what you think.

Reply:
Indeed, that doesn't print nicely